python/lsst/ip/isr/isrFunctions.py

```diff
@@ -597,13 +597,14 @@ def brighterFatterCorrection(exposure, kernel, maxIter, threshold, applyGain, ga
         kernelImage = afwImage.ImageD(kLx, kLy)
         kernelImage.getArray()[:, :] = kernel
         tempImage = image.clone()
+        tempImage = afwImage.ImageD(image.clone(), deep=True)
 
         nanIndex = numpy.isnan(tempImage.getArray())
         tempImage.getArray()[nanIndex] = 0.
 
-        outImage = afwImage.ImageF(image.getDimensions())
-        corr = numpy.zeros_like(image.getArray())
-        prev_image = numpy.zeros_like(image.getArray())
+        outImage = afwImage.ImageD(image.getDimensions())
+        corr = numpy.zeros_like(image.getArray(), dtype=numpy.float64)
+        prev_image = numpy.zeros_like(image.getArray(), dtype=numpy.float64)
```

Comment:
These two lines should be `image.array` (the modern API)

Reply:
Or maybe (similar to below) `numpy.zeros(image.array.shape, dtype=numpy.float64)`.